### ISSUE TITLE
Gradient Noise Implementation Design Doc

### ISSUE THREAD

**Issue by andrei3131** [lsds/KungFu#49]
![GradientNoise](https://user-images.githubusercontent.com/15161179/56848257-90084400-68de-11e9-8dda-cb73857c3218.png)

To concretely implement the above approach, we will use python. The computation of the gradient norm uses the locally computed gradient, the negotiated gradient via KungFu Parallel SGD, the batch size per worker and the global batch size.

We will use TensorFlow operators to calculate the biased estimates *at each iteration*:
-  To calculate the norm of the gradient tensor, we use `tf.norm()`, which treats the tensor as a vector and computes the L2 norm.
- All other variables in the formulas can be represented by `tf.constant()`
- Computing the exponentially moving average constitutes only from tensor multiplication by scalar and tensor additions.
- The noise scale is a `tf.constant()` at the end of the computation

DoD:
- Obtain the noise scale `tf.constant()`
